pkg/statsd/handler_http_forwarder_v2.go

```diff
@@ -395,7 +403,7 @@ func (hfh *HttpForwarderHandlerV2) serializeAndCompress(message proto.Message) (
 	return buf.Bytes(), nil
 }
 
-func (hfh *HttpForwarderHandlerV2) constructPost(ctx context.Context, logger logrus.FieldLogger, path string, message proto.Message) (func() error /*doPost*/, error) {
+func (hfh *HttpForwarderHandlerV2) constructPost(ctx context.Context, logger logrus.FieldLogger, path string, message proto.Message, dynHeaderTags string) (func() error /*doPost*/, error) {
```

Comment:
It would be better to avoid splinting the string on each construct post call, do we expected these dynamic header keys to change once the application is started?

Reply:
`dynHeaderTags` includes both tag names and values. The values cannot be decided by gostatsd itself. In the case of deploying gostatsd as a daemonset, it could receive metrics from multiple pods.

Reply:
@MovieStoreGuy @tiedotguy  `dynHeaderTags` in its string form is used as keys for the map of metricsMap splited by tags. The value is also calculated by `tagMatches` when iterating the metricMap.

Ideally the `tagMatch()` helper function should return `[][2]string` to perserve order of tag names. And `splitByTags()` returns `map[[][2]string]*MetricMap`, but slices/map cannot be used as map keys, at least without some gymnastics types. 

Considering this is forwarder and the code is only executed when `dynamic-headers` is enabled, I'm reluctant to implement such optimization.